Makefile

```
@@ -0,0 +1,71 @@
+# .DEFAULT_GOAL := all
+sources = src test
+
+.PHONY: .uv  ## Check that uv is installed
+.uv:
+	@uv -V || echo 'Please install uv: https://docs.astral.sh/uv/getting-started/installation/'
+
+.PHONY: install  ## Install the package, dependencies, and pre-commit for local development
+install: .uv
+	uv sync --frozen --group all --all-extras
+
+.PHONY: rebuild-lockfiles  ## Rebuild lockfiles from scratch, updating all dependencies
+rebuild-lockfiles: .uv
+	uv lock --upgrade
+
+.PHONY: format  ## Auto-format python source files
+format: .uv
+	uv run ruff format $(sources)
+
+.PHONY: check-format  ## Auto-format python source files
+check-format: .uv
+	uv run ruff format --check $(sources)
+
+.PHONY: lint  ## Lint python source files
+lint: .uv
+	uv run ruff check $(sources)
+
+.PHONY: typing  ## Run the type-checker
+typing: .uv
+	uv run mypy $(sources)
+
+.PHONY: test  ## Run all tests
+test: .uv
+	@uv run coverage run -m pytest --durations=10
+	@uv run coverage report
+
+.PHONY: benchmark  ## Run all benchmarks
+benchmark: .uv
+	uv run coverage run -m pytest --durations=10 --benchmark-enable tests/benchmarks
+
+.PHONY: testcov  ## Run tests and generate a coverage report, skipping the type-checker integration tests
+testcov: test
+	@echo "building coverage html"
+	@uv run coverage html
+
+.PHONY: all  ## Run the standard set of checks performed in CI
+all: lint format typing typing testcov
```

"typing" is repeated twice here
```suggestion
all: lint format typing testcov
```